numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -530,19 +530,32 @@ npy_remainder@c@(@type@ a, @type@ b)
 
 NPY_INPLACE @type@
 npy_floor_divide@c@(@type@ a, @type@ b) {
-    @type@ div, mod;
+    @type@ mod;
     if (NPY_UNLIKELY(!b)) {
+        if(npy_isinf(a)) {
+            return NPY_NAN;
+        }
```

Comment:
Why do you need this branch?  It is incorrect because it doesn't flag the floating point error ("invalid value encountered in division"); those warnings/errors may be tedious and not super useful, but there is hardly a reason to side-step it.

Reply:
```
import numpy as np
print(np.floor_divide(3, 0)) # returns 0
print(np.floor_divide(np.inf, 0)) # returns inf, should return nan
```

Reply:
the result is incorrect if the dividend is np.inf

Reply:
Yes, but again, normal division is correct:
```
In [6]: np.divide(np.inf, 0)
Out[6]: inf
```

Reply:
Wait, yes, it returns inf and not NaN.  But why do you expect a NaN here?

Reply:
following the description in the ticket #23860, that states that according to IEE 754 standard, 
'If the divisor is 0 and the dividend is infinite, the result is NaN (not a number).'

Reply:
`0/inf` is clearly NaN but not `inf/0`...  I think the author of that issue seems to have consistently swapped "divisor" and "dividend" and I can't blame them?

Reply:
You can trust the normal division to be IEEE compliant.  Any problem is about where floor division should or should not diverge from the IEEE description.

Reply:
ok, fixed